src/theory/bv/proof_checker.cpp

```diff
@@ -51,7 +63,27 @@ Node BVProofRuleChecker::checkInternal(PfRule id,
     Assert(args[0].getKind() == kind::BITVECTOR_EAGER_ATOM);
     return args[0].eqNode(args[0][0]);
   }
-  // no rule
+
+#define BV_PROOF_CASE(rule, name) \
```

Comment:
It cannot for the same reason why the `tryRule`s in `post_prove` cannot be inline functions.

Reply:
In the case of `postProve` you do not want to return a value in case that `tryRule` is false. Here you always return so I don't quite see the comparision.

Reply:
The logic here is like this:
```c++
else if (...) {
  return;
}
```
so if you factor out this entire else if into its own function, it would be something like
```c++
return some_function(...);
```
I don't see how this can not always return.

Reply:
In postProve this is the macro function:

```
#define POST_PROVE_CASE(name) \
    if (tryRule(cdp, eq, PfRule::name, {eq[0]})) \
    { \
      Trace("trewrite-rcons") << "Reconstruct " << eq << " (from " << tid << ", " \
                              << mid << ")"  << std::endl; \
      return true; \
    } \
    /* end of macro */
```

As we discussed in another comment, if the if condition  `tryRule(cdp, eq, PfRule::name, {eq[0]})` is false this does not return anything which makes it possible to write later in the function:

```
POST_PROVE_CASE(BV_UMULO_ELIMINATE)
POST_PROVE_CASE(BV_SMULO_ELIMINATE)
```

without a test whether `POST_PROVE_CASE(BV_UMULO_ELIMINATE)` was not true.